doc/usage/restructuredtext/directives.rst

```diff
@@ -483,72 +491,98 @@ __ http://pygments.org/docs/lexers/
    :rst:dir:`highlight` directive will be used.  If not set,
    :confval:`highlight_language` will be used.
 
-   **Additional options**
+   .. versionchanged:: 2.0
+      The ``language`` argument becomes optional.
 
-   Pygments can generate line numbers for code blocks. To enable this for, use
-   the ``linenos`` flag option. ::
+   .. rubric:: options
 
-      .. code-block:: ruby
-         :linenos:
+   .. rst:directive:option:: linenos
+      :type: no value
 
-         Some more Ruby code.
+      Enable to generate line numbers for the code block::
 
-   The first line number can be selected with the ``lineno-start`` option.  If
-   present, ``linenos`` flag is automatically activated::
+         .. code-block:: ruby
+            :linenos:
 
-      .. code-block:: ruby
-         :lineno-start: 10
+            Some more Ruby code.
 
-         Some more Ruby code, with line numbering starting at 10.
+   .. rst:directive:option:: lineno-start: number
+      :type: number
 
-   Additionally, an ``emphasize-lines`` option can be given to have Pygments
-   emphasize particular lines::
+      Set the first line number of the code block.  If present, ``linenos``
+      option is also automatically activated::
 
-    .. code-block:: python
-       :emphasize-lines: 3,5
+         .. code-block:: ruby
+            :lineno-start: 10
 
-       def some_function():
-           interesting = False
-           print 'This line is highlighted.'
-           print 'This one is not...'
-           print '...but this one is.'
+            Some more Ruby code, with line numbering starting at 10.
 
-   A ``caption`` option can be given to show that name before the code block.
-   A ``name`` option can be provided implicit target name that can be
-   referenced by using :rst:role:`ref`.  For example::
+      .. versionadded:: 1.3
 
-     .. code-block:: python
-        :caption: this.py
-        :name: this-py
+   .. rst:directive:option:: emphasized-lines: line numbers
+      :type: comma separated numbers
 
-        print 'Explicit is better than implicit.'
+      Empahsize particular lines of the code block::
```

Comment:
maybe we can use a code-block with reST lexer and emphasize lines number 6 and 8 ...
~~~rest
   Additionally, an ``emphasize-lines`` option can be given to have Pygments
   emphasize particular lines (we are showing output effect on top of input...):

   .. code-block:: rest
      :emphasize-lines: 6,8

      .. code-block:: python
         :emphasize-lines: 3,5

         def some_function():
             interesting = False
             print 'This line is highlighted.'
             print 'This one is not...'
             print '...but this one is.'
~~~

Reply:
With this change, it will be shown as:
![スクリーンショット 2019-12-10 1 25 51](https://user-images.githubusercontent.com/748828/70453429-15e2a380-1aec-11ea-8c6b-e6a4c3bc404a.png)

I feel emphasize input source code is strange a bit. But it might help users to imagine its behavior.

Reply:
I agree it looks a bit strange (with the highlighting going to the left of `def`). -0 to add this change...